test/persistence-hooks.suite.js

```diff
@@ -1646,6 +1891,44 @@ module.exports = function(dataSource, should) {
           });
       });
 
+      it('triggers `loaded` hook', function(done) {
+        TestModel.observe('loaded', pushContextAndNext());
+
+        TestModel.updateAll(
+          { name: existingInstance.name },
+          { name: 'changed' },
+          function(err, instance) {
+            if (err) return done(err);
+
+            observedContexts.should.eql(aTestModelCtx({
+              data: { name: 'changed' }
+            }));
+
+            done();
+          });
+      });
+
+      // TODO (pradnya), here as well `pushContextAndNext()` gets called through
+      // the notifyObserver() function twice. Why so?
```

Comment:
Hmm, that's surprising. Unless the second notification is caused by `loadTestModel`?

IMO, `updateAll` should not trigger the "loaded" hook at all, since it is not loading any model instances. The `updateAll` callback receives just an `info` object with a number of rows modified.